101-azurefirewall-premium/azuredeploy.json

```diff
@@ -0,0 +1,256 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                "description": "Deployment location"
+            }
+        }
+    },
+    "variables": {
```

Comment:
it looks like most of these variables need to be parameters, as they are affected by settings in the prereqs.  you will also need to add the resourceGroup name for the keyvault and the names of vnets/subnets

Reply:
Moving to single azuredeploy.